pkg/3scale/amp/operator/system_reconciler.go

```diff
@@ -373,6 +405,89 @@ func (r *SystemReconciler) systemAppDeploymentResourceMutator(desired, existing
 	return update, nil
 }
 
+func (r *SystemReconciler) systemZyncEnvVarMutator(desired, existing *k8sappsv1.Deployment) (bool, error) {
+	update := false
+
+	// Reconcile ZYNC_AUTHENTICATION_TOKEN env var
+	for idx := range existing.Spec.Template.Spec.Containers {
+		tmpChanged := helper.EnvVarReconciler(
+			desired.Spec.Template.Spec.Containers[idx].Env,
+			&existing.Spec.Template.Spec.Containers[idx].Env,
+			"ZYNC_AUTHENTICATION_TOKEN")
+		update = update || tmpChanged
+	}
+
+	return update, nil
+}
+
+func (r *SystemReconciler) routesExist() (bool, error) {
+	expectedRoutes := 6
+	opts := k8sclient.ListOptions{
+		Namespace: r.apiManager.GetNamespace(),
+	}
+
+	routes := routev1.RouteList{}
+	err := r.Client().List(r.Context(), &routes, &opts)
+	if err != nil {
+		return false, err
+	}
+
+	if len(routes.Items) < expectedRoutes {
+		r.Logger().Info(fmt.Sprintf("Missing routes - expected at least %d routes but only found %d", expectedRoutes, len(routes.Items)))
+		return false, nil
+	}
+	return true, nil
+}
+
+func (r *SystemReconciler) isZyncReady() (bool, error) {
+	zyncDeploymentNames := []string{"zync", "zync-database", "zync-que"}
+
+	for _, deploymentName := range zyncDeploymentNames {
+		deployment := &k8sappsv1.Deployment{}
+		err := r.Client().Get(r.Context(), k8sclient.ObjectKey{
+			Namespace: r.apiManager.GetNamespace(),
+			Name:      deploymentName,
+		}, deployment)
+
+		// Return error if can't get deployment
+		if err != nil && !k8serr.IsNotFound(err) {
+			return false, fmt.Errorf("error getting deployment %s: %w", deployment.Name, err)
+		}
```

Comment:
for all this..can we not just rely on apim deployment status where zync is either ready/starting or stopped?

Reply:
We can but I figured fetching the Deployments directly was more accurate in case the APIManager .status hadn't been updated yet in the reconciliation flow